vllm/sequence.py

```diff
@@ -1044,19 +1048,36 @@ class HiddenStates:
     seq_ids are the sequence ids of each entry of the batch
     dimension of the hidden_states tensor"""
 
-    def __init__(self, seq_group_metadata_list: List[SequenceGroupMetadata],
-                 hidden_states: torch.Tensor):
+    def __init__(
+            self,
+            seq_group_metadata_list: List[SequenceGroupMetadata],
+            hidden_states: torch.Tensor,
+            last_non_bonus_token_hidden_states: Optional[torch.Tensor] = None):
```

Comment:
Add a docstring for the new field? seems non obvious

Reply:
Also renaming last_non_bonus_token_hidden_states to second_last_token_hidden_states as that is more appropriate.